### PR TITLE
feat(swap,sdk): [WEB-343] add ccmGasBudget and ccmMessage to Swap

### PR DESCRIPTION
**Pull request by jerryafr**
[WEB-343]
- add ccmGasBudget and ccmMessage to Swap

### CODE REVIEW

File: packages/sdk/src/swap/types.ts [hunk 1]
```diff
@@ -58,6 +58,8 @@ export interface CommonStatusFields extends ChainsAndAssets {
   depositChannelCreatedAt: number | undefined;
   expectedDepositAmount: string | undefined;
   depositChannelExpiryTime: number | undefined;
+  ccmGasBudget: string | undefined;
```
Comment (niklasnatter) [inline packages/sdk/src/swap/types.ts:61]: i think i would bundle this into a `ccmMetadata` object to be consistent with the params of `requestDepositAddress`/`executeSwap `

File: packages/swap/src/event-handlers/__tests__/__snapshots__/ccmSwap.test.ts.snap [hunk 2]
```diff
@@ -33,6 +33,9 @@ exports[`batch swap flow handles all the events: egress 1`] = `
 
 exports[`batch swap flow handles all the events: gas swap 1`] = `
 {
+  "ccmDepositReceivedBlockIndex": null,
```
Comment (niklasnatter) [inline packages/swap/src/event-handlers/__tests__/__snapshots__/ccmSwap.test.ts.snap:36]: shouldnt this be set at some point in this test? 🤔

File: packages/swap/src/event-handlers/__tests__/ccmDepositReceived.test.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        },
+      });
+    });
+
+    const swap = await prisma.swap.findFirstOrThrow({
+      where: { nativeId: BigInt(9876545) },
+    });
+
+    expect(swap.ccmGasBudget?.toString()).toEqual('65000');
+    expect(swap.ccmMessage).toEqual('some-random-text');
```
Comment (niklasnatter) [inline packages/swap/src/event-handlers/__tests__/ccmDepositReceived.test.ts:63]: we can check the `ccmDepositReceivedBlockIndex` here too

File: packages/swap/src/routes/__tests__/swap.test.ts [hunk 4]
```diff
@@ -522,6 +536,8 @@ describe('server', () => {
       expect(BigInt(swapId)).toEqual(nativeId);
       expect(rest).toMatchInlineSnapshot(`
         {
+          "ccmDepositReceivedBlockIndex": null,
```
Comment (niklasnatter) [inline packages/swap/src/routes/__tests__/swap.test.ts:539]: maybe we can add a test with metadata for this endpoint

File: packages/swap/src/event-handlers/ccmDepositReceived.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import { encodedAddress } from './common';
+import type { EventHandlerArgs } from '.';
+
+const ccmChannelMetadataArgs = z.object({
+  message: z.string(),
+  gasBudget: u128,
+});
+
+const ccmDepositMetadataArgs = z.object({
+  channelMetadata: ccmChannelMetadataArgs,
```
Comment (zoheb391) [inline packages/swap/src/event-handlers/ccmDepositReceived.ts:12]:
```suggestion
  channelMetadata:  z.object({
    message: z.string(),
    gasBudget: u128,
  })
```

**Review by zoheb391 — APPROVED** [review]

**Review by niklasnatter — APPROVED** [review]

**Pull request merged**